streaming/src/main/scala/org/apache/spark/streaming/DStreamGraph.scala

```diff
@@ -29,8 +28,8 @@ import org.apache.spark.util.Utils
 
 final private[streaming] class DStreamGraph extends Serializable with Logging {
 
-  private val inputStreams = new ArrayBuffer[InputDStream[_]]()
-  private val outputStreams = new ArrayBuffer[DStream[_]]()
+  private var inputStreams = Array.empty[InputDStream[_]]
```

Comment:
What about using mutable.Seq? does that also work?
This probably is OK to use an Array as setting the streams isn't called many times though

Reply:
Let me try with mutable.Seq, will feedback later

Reply:
- default implementation of `mutable.Seq` is `ArrayBuffer`, it can't work.

- default implementation of `immutable.Seq` is `List`, it can't work also

- `mutable.ArraySeq` and `immutable.ArraySeq` both work, but still need use `var` because  they are not `Growable`

Reply:
recommend `mutable.ArraySeq` or `immutable.ArraySeq`? Or keep  use `Array`?

Reply:
mutable.ArraySeq sounds best, if that exists in 2.12 and 2.13.

Reply:
Ok ~ Address 3fef228 fix this